### PR TITLE
Fix for invalid analysis when cljs macros expanded

### PR DESCRIPTION
**Pull request by rvlo**
It was causing all sorts of weird behaviour, including broken linting

### CODE REVIEW

**Comment by rvlo** [comment]
In the event of an exception, the go-loop was breaking and never again analyzing changes.

To reproduce: insert an unbalanced close parenthesis. The output:

<pre>
                                              java.lang.Thread.run              Thread.java:  834
                                                               ...
clojure.core.async.impl.concurrent/counted-thread-factory/reify/fn           concurrent.clj:   29
                java.util.concurrent.ThreadPoolExecutor$Worker.run  ThreadPoolExecutor.java:  628
                 java.util.concurrent.ThreadPoolExecutor.runWorker  ThreadPoolExecutor.java: 1128
                                                               ...
             clojure.core.async.impl.channels.ManyToManyChannel/fn             channels.clj:  135
                       clojure.core.async.impl.ioc-macros/take!/fn           ioc_macros.clj:  991
      clojure.core.async.impl.ioc-macros/run-state-machine-wrapped           ioc_macros.clj:  982
              clojure.core.async.impl.ioc-macros/run-state-machine           ioc_macros.clj:  978
                     clojure-lsp.main/run/fn/state-machine--auto--                 main.clj:  397
                  clojure-lsp.main/run/fn/state-machine--auto--/fn                 main.clj:  397
               clojure-lsp.feature.file-management/analyze-changes      file_management.clj:  128
                      clojure-lsp.feature.diagnostics/lint-project          diagnostics.clj:   94
                  clojure-lsp.feature.diagnostics/find-diagnostics          diagnostics.clj:   86
       clojure-lsp.feature.diagnostics/kondo-findings->diagnostics          diagnostics.clj:   39
                                                 clojure.core/mapv                 core.clj: 6905
                                               clojure.core/reduce                 core.clj: 6830
                                       clojure.core.protocols/fn/G            protocols.clj:   13
                                         clojure.core.protocols/fn            protocols.clj:   75
                                 clojure.core.protocols/seq-reduce            protocols.clj:   31
                                       clojure.core.protocols/fn/G            protocols.clj:   19
                                         clojure.core.protocols/fn            protocols.clj:  136
                                                               ...
                                              clojure.core/mapv/fn                 core.clj: 6914
    clojure-lsp.feature.diagnostics/kondo-findings->diagnostics/fn          diagnostics.clj:   39
         clojure-lsp.feature.diagnostics/kondo-finding->diagnostic          diagnostics.clj:   15
                                     clojure-lsp.parser/loc-at-pos               parser.clj:   93
                            clojure-lsp.parser/safe-zloc-of-string               parser.clj:   74
                                    rewrite-clj.zip.base/of-string                 base.clj:   71
                                    rewrite-clj.zip.base/of-string                 base.clj:   73
                               rewrite-clj.parser/parse-string-all               parser.clj:   35
                                      rewrite-clj.parser/parse-all               parser.clj:   20
                                                clojure.core/doall                 core.clj: 3136
                                                clojure.core/dorun                 core.clj: 3130
                                                 clojure.core/next                 core.clj:   64
                                                               ...
                                        clojure.core/take-while/fn                 core.clj: 2908
                                                  clojure.core/seq                 core.clj:  139
                                                               ...
                                        clojure.core/repeatedly/fn                 core.clj: 5138
                                   rewrite-clj.parser/parse-all/fn               parser.clj:   18
                                          rewrite-clj.parser/parse               parser.clj:   13
                                rewrite-clj.parser.core/parse-next                 core.clj:   35
                                 rewrite-clj.reader/read-with-meta               reader.clj:  132
                                                               ...
                              rewrite-clj.parser.core/eval32981/fn                 core.clj:   71
                                                               ...
                                   rewrite-clj.reader/throw-reader               reader.clj:   11
                                   rewrite-clj.reader/throw-reader               reader.clj:   18
java.lang.Exception: Unmatched delimiter: ) [at line 6, column 1]
</pre>

File: src/clojure_lsp/main.clj [hunk 1]
```diff
@@ -395,7 +395,9 @@
         (producer/publish-diagnostic (<! debounced-diags))
         (recur))
       (go-loop []
-        (f.file-management/analyze-changes (<! debounced-changes))
+        (try
+          (f.file-management/analyze-changes (<! debounced-changes))
+          (catch Exception e (log/error e "in analyze go-loop")))
```
Comment (ericdallo) [inline src/clojure_lsp/main.clj:400]: Could we improve a little better the error message?
Maybe
```suggestion
          (catch Exception e 
            (log/error e "Error during analyzing buffer file changes")))
```

**Review by ericdallo — APPROVED** [review]
Thank you for the fix, just a suggestion about the log name, but LGTM

**Comment by rvlo** [comment]
Of course! My pleasure!

**Comment by ericdallo** [comment]
Thank you!

**Pull request merged**